src/python/scripts/dx-mount-all-inputs

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python
+#
+# Copyright (C) 2014-2016 DNAnexus, Inc.
+#
+# This file is part of dx-toolkit (DNAnexus platform client libraries).
+#
+#   Licensed under the Apache License, Version 2.0 (the "License"); you may not
+#   use this file except in compliance with the License. You may obtain a copy
+#   of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#   WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#   License for the specific language governing permissions and limitations
+#   under the License.
+
+import argparse
+import dxpy
+from dxpy.utils.printing import fill, refill_paragraphs, BOLD, RED
+
+description = BOLD('Note') + ''': this is a utility for use by bash apps
+running in the DNAnexus Platform.
+
+Mounts all files that were supplied as inputs to the app.  By
+convention, if an input parameter "FOO" has value
+
+    {"$dnanexus_link": "file-xxxx"}
+
+and filename INPUT.TXT, then the linked file will be mounted into the
+path:
+
+    $HOME/in/FOO/INPUT.TXT
+
+If an input is an array of files, then all files will be placed into
+numbered subdirectories under a parent directory named for the
+input. For example, if the input key is FOO, and the inputs are {A, B,
```

Comment:
When multiple files in the array have the same name, e.g. {A, A, A}.vcf, will the target directory structure look the same?

Reply:
Those same file names end up getting mounted in unique locations, just like dx-download-all-input downloads them into unique locations. So for an array of files that get mounted it would look like this:

/home/dnanexus/in/csv/0/newdata1.csv
/home/dnanexus/in/csv/1/newdata1.csv
/home/dnanexus/in/csv/2/newdata2.csv